### PR TITLE
 Fix FiniteElement::has_generalized_support_points() and FiniteElement::has_face_support_points()  for FE_Nothing. #17584 

### PR DESCRIPTION
**Pull request by OresteMarquis**
This is an extension of the work in the PR:  Fix FiniteElement::has_support_points() for FE_Nothing. #17584 

I tried to generalize what is done in the above PR, but I may have made some mistakes. Also, I wasn't sure if I had to modify the function in the fe.cc "FiniteElement::has_support_on_face()", since I was not sure of its purpose.

@bangerth

### CODE REVIEW

**Review by bangerth — COMMENTED** [review]
Yes, this looks right, but the spell checker complains about your changelog entry.

Separately, would you please drop the merge commit from master? If at all possible, merging from master should be avoided. (If you need to, you can "rebase" onto master."

**Comment by OresteMarquis** [comment]
Is there something else that I can do @bangerth , or something I didi wrong?

File: doc/news/changes/minor/20240821Marquis [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Fixed: Extended the fix for the FiniteElement::has_support_points() function. 
+Now, FiniteElement::has_generalized_support_points() and 
+FiniteElement::has_face_support_points() are also corrected. Previously, these
+functions returned false for the FE_Nothing element. Now, FE_Nothing::has_generalized_support_points() and 
+FE_Nothing::has_face_support_points() correctly return true, as the empty 
+arrays returned by FE_Nothing::get_generalized_support_points() and FE_Nothing::get_unit_face_support_points() accurately describe the support 
+points of the element(i.e., they don't have any, as there are no
+degrees of freedom).
```
Comment (masterleinad) [inline doc/news/changes/minor/20240821Marquis:8]:
```suggestion
Fixed: Previsouly, FiniteElement::has_generalized_support_points() and 
FiniteElement::has_face_support_points() returned false for the FE_Nothing element. Now, FE_Nothing::has_generalized_support_points() and 
FE_Nothing::has_face_support_points() correctly return true, as the empty 
arrays returned by FE_Nothing::get_generalized_support_points() and FE_Nothing::get_unit_face_support_points() accurately describe the support 
points of the element(i.e., they don't have any, as there are no
degrees of freedom).
```

**Review by masterleinad — APPROVED** [review]

File: doc/news/changes/minor/20240821Marquis [hunk 2]
```diff
@@ -1,7 +1,5 @@
-Fixed: Extended the fix for the FiniteElement::has_support_points() function. 
-Now, FiniteElement::has_generalized_support_points() and 
-FiniteElement::has_face_support_points() are also corrected. Previously, these
-functions returned false for the FE_Nothing element. Now, FE_Nothing::has_generalized_support_points() and 
+Fixed: Previsouly, FiniteElement::has_generalized_support_points() and
```
Comment (masterleinad) [inline doc/news/changes/minor/20240821Marquis:1]:
```suggestion
Fixed: Previously, FiniteElement::has_generalized_support_points() and 
```

**Comment by bangerth** [comment]
You did nothing wrong other than the fact that I was out of town for the weekend :-)

The indentation checker still fails:
```
Invalid author 'OresteMarquis' without firstname and lastname

For possible solutions, consult the following page:
      https://github.com/dealii/dealii/wiki/Commit-authorship
```

**Review by masterleinad — APPROVED** [review]

**Comment by bangerth** [comment]
The `fe_nothing_dominates_03` test is now failing with this error:
```

--------------------------------------------------------
An error occurred in line <644> of file </jenkins/workspace/dealii-serial_PR-17590/source/fe/fe_q_base.cc> in function
    void dealii::FE_Q_Base<dim, spacedim>::get_subface_interpolation_matrix(const dealii::FiniteElement<dim, spacedim>&, unsigned int, dealii::FullMatrix<double>&, unsigned int) const [with int dim = 2; int spacedim = 2]
The violated condition was: 
    interpolation_matrix.n() == this->n_dofs_per_face(face_no)
Additional information: 
    Two sizes or dimensions were supposed to be equal, but aren't. They
    are 0 and 2.
--------------------------------------------------------
```
Does that look familiar? Would you be willing to debug what is happening here?

**Comment by OresteMarquis** [comment]
I am not familiar with this error, but i can try to figure it out!

**Comment by blaisb** [comment]
> I am not familiar with this error, but i can try to figure it out!

If you need help tell me i'll give you a hand with this.

**Comment by OresteMarquis** [comment]
I am not sure it is the best solution I implemented, for the fe_nothing_dominates_03 fix, but it's the only thing I could come up with.

File: doc/news/changes/minor/20240821Marquis [hunk 3]
```diff
@@ -0,0 +1,8 @@
+Fixed: Previously, FiniteElement::has_generalized_support_points() and 
+FiniteElement::has_face_support_points() returned false for the FE_Nothing element. Now, FE_Nothing::has_generalized_support_points() and 
+FE_Nothing::has_face_support_points() correctly return true, as the empty 
+arrays returned by FE_Nothing::get_generalized_support_points() and FE_Nothing::get_unit_face_support_points() accurately describe the support 
+points of the element(i.e., they don't have any, as there are no
+degrees of freedom).
+<br>
+(Oreste Marquis, 2024/08/21)
```
Comment (bangerth) [inline doc/news/changes/minor/20240821Marquis:8]: github says that you need an endline here. (In fact, every file should always end in a newline.)

**Review by bangerth — COMMENTED** [review]
Yes, this looks good with the exception of the one missing newline.

Could I ask you to merge the commits into one? Or one each for code changes/changelog/new test?

**Comment by bangerth** [comment]
Thanks again for working on this! We very much appreciate the contribution!

**Comment by OresteMarquis** [comment]
I merge everything into one commit since changes/changelog/newtest were made in the same commit previously. Is the global description of the commit ok?

Thank you for guiding me in the pr process :) It really helped!

**Review by bangerth — APPROVED** [review]

**Pull request merged**